### PR TITLE
Accessing a URL with invalid branch results in 404 Not found

### PR DESCRIPTION
**Pull request by harshm-dev**
Created a page for NotFound, API 404 during commit log/entries will now result in a redirection to 404 page.

### CODE REVIEW

**Comment by harshm-dev** [comment]
notfound

File: ui/src/Explore/CommitLog.tsx [hunk 1]
```diff
@@ -61,13 +61,17 @@ const CommitLog = (props: {
...
     const logs = async () => {
-      const results = await fetchLog(currentRef);
-      if (results) {
-        setLogList(results);
+      const fetchLogResults = await fetchLog(currentRef);
+      if (fetchLogResults) {
+        setLogList(fetchLogResults);
+        setRefNotFound(false);
+      } else if (fetchLogResults === undefined) {
+        setRefNotFound(true);
```
Comment (nastra) [inline ui/src/Explore/CommitLog.tsx:74]: nit: intead of the if-else, maybe just `setRefNotFound(fetchLogResults === undefined)`

**Review by nastra — DISMISSED** [review]
LGTM with one nit

**Comment by harshm-dev** [comment]
@nastra can you please review?

**Review by nastra — APPROVED** [review]

**Pull request merged**